myia/simplify_types.py

```diff
@@ -493,7 +493,10 @@ def from_canonical(self, arg, orig_t: AbstractUnion):
 
 @ovld  # noqa: F811
 def from_canonical(self, arg, orig_t: AbstractType):
-    return arg
+    if isinstance(orig_t.element, AbstractHandle):
+        return HandleInstance
+    else:
+        return arg
```

Comment:
Why is it necessary to have a special case here?

Reply:
Indeed, useless. I just needed to fix HandleInstance conversion in Pytorch backend, instead.